File: Exavision.Seasense.Shared/Settings/SettingMaterial.cs

using System.Collections.Generic;


namespace Exavision.Seasense.Shared.Settings {

    public enum MaterialType {
        DayCamera,
        ThermalCamera,
        Turret,
        LazerMeasurement,
        InertialMeasurement,
        Unit

    }
    public abstract class SettingMaterial : SettingBase {
        public MaterialType MaterialType { get; set; }
        public string DisplayName { get; set; }
        public List<SettingCapability> Capabilities { get; set; } = new List<SettingCapability>();
        public List<SettingMaterial> Materials { get; set; } = new List<SettingMaterial>();
    }
}

File: Exavision.Seasense/Client/src/app/materials/settings/setting-material.ts
import { MaterialType } from "../material-type";
import { SettingCapability } from "./setting-capability";


export interface SettingMaterial {
  id: string;  
  materialType: MaterialType;
  materials: SettingMaterial[];
  capabilities: SettingCapability[];
  type: string;
  displayName: string;
}


